### PR TITLE
add support for Instana and switch New Relic to use enable flag like …

### PR DESCRIPTION
**Pull request by gchickma**
Add support for Instana.  Only app configuration required - Instana auto discovers all services.

Also switch New Relic to use enable flag like all other monitoring options

### CODE REVIEW

File: bmrg-flow/templates/deployment-service.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@ spec:
       {{- if and (eq $k "controller") ($values.workers.security.enable) }}
       serviceAccountName: {{ include "bmrg.name" (dict "context" $context "component" "controller" ) }}
       {{- end }}
-      {{- if $.Values.monitoring.newrelic.apm.key }}
+      {{- if $.Values.monitoring.newrelic.enable }}
```
Comment (tlawrie) [inline bmrg-flow/templates/deployment-service.yaml:30]: Would this have to be an AND... i.e. if its enabled and there is an APM server key?

File: bmrg-flow/templates/deployment-service.yaml [hunk 2]
```diff
@@ -42,13 +42,15 @@ spec:
           imagePullPolicy: "{{ $values.image.pullPolicy }}"
           env:
           - name: JAVA_OPTS
-            value: '-Dspring.config.additional-location=file:/data/application.properties{{- if $.Values.monitoring.newrelic.apm.key }}{{- if $.Values.proxy.enable }} -Dnewrelic.config.proxy_host={{ $.Values.proxy.host }} -Dnewrelic.config.proxy_port={{ $.Values.proxy.port }} -Dnewrelic.config.distributed_tracing.enabled=true{{- end }} -javaagent:/opt/newrelic/newrelic.jar{{- end }}{{- if eq $k "workflow" }} -Dmongo.encrypt.secret=${FLOW_ENCRYPT_SECRET}  -Dmongo.encrypt.salt=${FLOW_ENCRYPT_SALT}{{- end }}'
+            value: '-Dspring.config.additional-location=file:/data/application.properties{{- if $.Values.monitoring.newrelic.enable }}{{- if $.Values.proxy.enable }} -Dnewrelic.config.proxy_host={{ $.Values.proxy.host }} -Dnewrelic.config.proxy_port={{ $.Values.proxy.port }} -Dnewrelic.config.distributed_tracing.enabled=true{{- end }} -javaagent:/opt/newrelic/newrelic.jar{{- end }}{{- if eq $k "workflow" }} -Dmongo.encrypt.secret=${FLOW_ENCRYPT_SECRET}  -Dmongo.encrypt.salt=${FLOW_ENCRYPT_SALT}{{- end }}'
```
Comment (tlawrie) [inline bmrg-flow/templates/deployment-service.yaml:45]: Does Instana need JVM params?

File: bmrg-flow/values.yaml [hunk 3]
```diff
@@ -141,7 +141,13 @@ proxy:
   port:
 
 monitoring:
+  instana:
+    enable: false
+    browser:
+      reportingUrl:
+      key:
   newrelic:
+    enable: false
```
Comment (tlawrie) [inline bmrg-flow/values.yaml:150]: Looks GTM

File: bmrg-flow/templates/deployment-service.yaml (hunk 1 above)
Comment (gchickma) [reply bmrg-flow/templates/deployment-service.yaml:30]: I guess so yeah, you could have New Relic enabled but you don't use APM.  I can update this.

File: bmrg-flow/templates/deployment-service.yaml (hunk 2 above)
Comment (gchickma) [reply bmrg-flow/templates/deployment-service.yaml:45]: Nope.  Instana auto-discovers services when running under K8S.  The only configuration required is for apps when running web page monitoring.

File: bmrg-flow/templates/deployment-service.yaml (hunk 1 above)
Comment (gchickma) [reply bmrg-flow/templates/deployment-service.yaml:30]: @tlawrie Probably also need to change to and AND on deployment-app.yaml too right?  Do we need to check monitoring.newrelic.browser.key ?

Comment (gchickma) [reply bmrg-flow/templates/deployment-service.yaml:30]: Eg. in deployment-app.yaml does the condition need to be:
`{{- if and ($.Values.monitoring.newrelic.enable) ($.Values.monitoring.newrelic.apm.key) ($.Values.monitoring.newrelic.browser.key) }}` ?

**Comment by timrbula** [comment]
@morarucostel Can you take a look at this when you have a moment?

**Review by morarucostel — APPROVED** [review]
LGTM, thank you @gchickma

File: bmrg-flow/templates/deployment-service.yaml (hunk 1 above)
Comment (gchickma) [reply bmrg-flow/templates/deployment-service.yaml:30]: Hey @tlawrie just checking back in with best way forward here.  Thanks.

Comment (tlawrie) [reply bmrg-flow/templates/deployment-service.yaml:30]: Yep. Probably. I can't quite remember what is in deployment yaml

Comment (gchickma) [reply bmrg-flow/templates/deployment-service.yaml]: @tlawrie I've made further updates to Instana and New Relic.  Please take a look when you can.  Thx.

**Review by tlawrie — APPROVED** [review]

**Comment by gchickma** [comment]
@tlawrie  I've made further updates for Instana and New Relic configuration. Please take a look when you can. Thx.

**Comment by tlawrie** [comment]
Hi @gchickma I have already approved this and GitHub isn't showing any new changes.

Screen Shot 2022-03-29 at 10 41 23 am

**Comment by timrbula** [comment]
@gchickma to follow up on the above, when you return can you make sure that your changes pushed to the remote?

**Comment by gchickma** [comment]
@timrbula All my changes are on remote branch - I see the changes here https://github.com/boomerang-io/charts/pull/62/files

**Comment by tlawrie** [comment]
hey @gchickma. We were on the community call trying to figure out why you were asking for a review. We thought you must have thought you pushed additional changes that needed review.

If there are no new changes, this PR already has my approval. You may need @morarucostel to approve as well?

**Review by morarucostel — APPROVED** [review]

**Pull request merged**